orchestrator/kafka/root.go

```diff
@@ -105,12 +106,15 @@ func (c *Component) Start() error {
 			l.Warn().Msgf("mismatch for replication factor: got %d, want %d",
 				topic.ReplicationFactor, c.config.TopicConfiguration.ReplicationFactor)
 		}
-		if err := admin.AlterConfig(sarama.TopicResource, c.kafkaTopic, c.config.TopicConfiguration.ConfigEntries, false); err != nil {
-			l.Err(err).Msg("unable to set topic configuration")
-			return fmt.Errorf("unable to set topic configuration for %q: %w",
-				c.kafkaTopic, err)
+		if !helpers.IsMapSubset(topic.ConfigEntries, c.config.TopicConfiguration.ConfigEntries) ||
+			(c.config.TopicConfiguration.ConfigEntriesStrictSync && helpers.Diff(topic.ConfigEntries, c.config.TopicConfiguration.ConfigEntries) != "") {
+			if err := admin.AlterConfig(sarama.TopicResource, c.kafkaTopic, c.config.TopicConfiguration.ConfigEntries, false); err != nil {
+				l.Err(err).Msg("unable to set topic configuration")
+				return fmt.Errorf("unable to set topic configuration for %q: %w",
+					c.kafkaTopic, err)
+			}
+			l.Info().Msg("topic updated")
```

Comment:
Add a function `shouldAlterConfiguration(target, source map[string]*string, strict bool) bool` and use it as a condition. Then, add unit tests for this function. Also, I would prefer to not use `helpers.Diff()`, nor an helper using reflection (they are hard to completely test), just write some ad-hoc code:
```go
for k, v := range target {
  if ov, ok := source[k]; !ok || *ov != *v {
    return true
  }
}
if !strict {
  return false
}
return len(target) != len(source)
```

Reply:
You're right, I'll keep it simple as suggested, thanks :)